person/src/homeworks/person/core/model/Singer.java

```diff
@@ -29,11 +40,76 @@ public String getBandName() {
 
   @Override
   public void singing() {
-    System.out.println("practising.");
+    if(full) {
+      System.out.printf("%s %s", name, surname + " is singing.");
+      System.out.println();
+      ((SingerAction) actions).singing();
+    }
+    else System.out.println("She/He hasn't ate, she/he can't sing.");
   }
 
   @Override
   public void playingGuitar() {
-    System.out.println("playing guitar.");
+    if(walked && full) {
+      System.out.printf("%s %s", name, surname);
+      System.out.println();
+      ((SingerAction) actions).playingGuitar();
+    }
+    else System.out.println("She/He hasn't walked and ate yet, she/he can't play.");
+  }
+
+  @Override
+  public void learn() {
+    System.out.printf("%s %s", name, surname + " is learning.");
+    System.out.println();
+    actions.learn();
+  }
+
+  @Override
+  public void eat() {
+    full = true;
+    System.out.printf("%s %s", name, surname + " is eating");
+    System.out.println();
+    actions.eat();
+  }
+
+  @Override
+  public void walk() {
+    walked = true;
+    System.out.printf("%s %s", name, surname + " is walking.");
+    System.out.println();
+    actions.walk();
+  }
+
+  @Override
+  public void setDesignation(String designation) {
+      this.designation = SingerType.valueOf(designation);
+  }
+
+  @Override
+  public String getDesignation() {
+    return designation.toString();
+  }
+
+  @Override
+  public void act(String actionType)
+  {
+    switch (actionType) {
+      case "eat":
```

Comment:
@tatevikb what about enumeration for this cases too ?

Reply:
Done.